lib/service/service.go

```diff
@@ -2273,6 +2270,7 @@ func (process *TeleportProcess) initSSH() error {
 		if err != nil {
 			return trace.Wrap(err)
 		}
+		defer func() { warnOnErr(s.Close(), log) }()
```

Comment:
@espadolini does this look safe?

If there's an error and we return early, this close will do an immediate shutdown.

If there's no error and everything starts up successfully, then the clean up at the end of this function will run, and my hope is that the deferred `Close` call will effectively be a no op.

Reply:
I think a double close on most of our networking stuff is either a noop or is harmless but returns a `net.ErrClosed` or equivalent (and thus gets ignored by `warnOnErr`) so we're good here.